src/ngraph/runtime/cpu/cpu_external_function.cpp

```diff
@@ -1447,7 +1438,8 @@ void runtime::cpu::CPU_ExternalFunction::build()
     pass_manager.register_pass<ngraph::pass::Liveness>();
     pass_manager.register_pass<ngraph::pass::PropagateCacheability>(
         runtime::cpu::get_annotations_factory());
-    pass_manager.register_pass<ngraph::pass::MemoryLayout>(size_t(s_memory_pool_alignment), true);
+    pass_manager.register_pass<runtime::cpu::pass::CPUMemoryAssignment>(
+        size_t(s_memory_pool_alignment), !m_pass_config.get_reuse_memory());
```

Comment:
should be pass_manager.get_pass_config().get_reuse_memory()

Reply:
pass_manager.get_pass_config() only gets the reuse memory flag from the environment variable. m_pass_config gets that flag from the environment variable and the overloaded compile() function.